lib/Dialect/FIRRTL/Transforms/SpecializeOption.cpp

```diff
@@ -79,12 +79,15 @@ struct SpecializeOptionPass
               target = inst.getTargetOrDefaultAttr(it->second);
 
             ImplicitLocOpBuilder builder(inst.getLoc(), inst);
+            // TODO: Fixup this to lower domains.
+            SmallVector<Attribute> domains;
             auto newInst = InstanceOp::create(
                 builder, inst->getResultTypes(), target, inst.getNameAttr(),
                 inst.getNameKindAttr(), inst.getPortDirectionsAttr(),
-                inst.getPortNamesAttr(), inst.getAnnotationsAttr(),
-                inst.getPortAnnotationsAttr(), builder.getArrayAttr({}),
-                UnitAttr{}, UnitAttr{}, inst.getInnerSymAttr());
+                inst.getPortNamesAttr(), builder.getArrayAttr(domains),
```

Comment:
Consider `getArrayAttr({})` (or doing this once and re-using) like other empty-array arguments.
With the TODO, maybe it's done this way to build on later so 🤷 .

Reply:
This is fixed in https://github.com/llvm/circt/pull/8843/commits/8465e1ed86a73f695b676748a97ce8cd322d2379.